homeassistant/components/otp/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow for One-Time Password (OTP) integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import pyotp
+import voluptuous as vol
+
+from homeassistant.auth.mfa_modules.totp import _generate_qr_code
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_CODE, CONF_NAME, CONF_TOKEN
+
+from .const import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_TOKEN): str,
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+    }
+)
+
+STEP_CONFIRM_DATA_SCHEMA = vol.Schema({vol.Required(CONF_CODE): str})
+
+
+class TOTPConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for One-Time Password (OTP)."""
+
+    VERSION = 1
+    user_input: dict[str, Any]
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            # generate new token if empty
+            if not user_input.get(CONF_TOKEN):
```

Comment:
Why are you generating a new token if it is not provided?
The yaml currently requires the token. 
We should not add new functionality in this PR. It can be added in a follow up PR to keep this PR as small as possible

Reply:
The documentation already describes the manual steps how to do this, so i thought the config flow can do this for the user. It is just 2 lines of code but i can add this functionality in a separate PR if necessary